Course3/Infrasrtucture/UserWorker.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Course3.Models;
+using Encoder;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Security;
+using System.Text;
+
+namespace Course3.Infrasrtucture
+{
+    class UserWorker
+    {
+        private string _login;
+        private SecureString _password;
+        private readonly string _dataPath = @"Resources\UsersData.txt";
+
+        public string Login { get => _login; private set => _login = value; }
+        public SecureString Password { get => _password; private set => _password = value; }
+
+        public UserWorker()
+        {
+            
+        }
+
+        public ObservableCollection<User> ReadUsers()
```

Comment:
Если это сервис, то лучше если он будет возвращать просто перечисление. ObservableCollection - это ближе к интерфейсу пользвоателя
```suggestion
        public IEnumerable<User> ReadUsers()
```